scan/lib/scan/runner.rb

```diff
@@ -165,6 +160,18 @@ def test_results
       File.read(Scan.cache[:temp_junit_report])
     end
 
+    def prelaunch_simulators
+      return unless Scan.devices.to_a.size > 0 # no devices selected, no sims to launch
+
+      # Return early unless the user wants to prelaunch simulators. Or if the user wants simulator logs
+      # and the sim OS version is >= 13, then we must prelaunch simulators because Xcode's headless
+      # mode launches and shutsdown the simulators before we can collect the logs.
+      os_at_least_13 = Gem::Version.new(Scan.devices.first.os_version) >= Gem::Version.new('13.0')
+      return unless Scan.config[:prelaunch_simulator] || (Scan.config[:include_simulator_logs] && os_at_least_13)
+
+      FastlaneCore::Simulator.launch(Scan.devices.first)
```

Comment:
I took a look at what `FastlaneCore::Simulator.launch` does and it's actually booting the GUI simulator. This is not what we want in order to collect logs. I expect it will slow down testing, and also just be annoying. Instead we want to run `xcrun simctl boot <UDID>` to boot it. We'll also want to check the current status prior to booting, in order to shut the simulator down again if we booted it (but keep it running if we didn't). Which is to say, we want to replicate Xcode's headless behavior.

And please do the shutdowns in a manner that will still occur even if an exception is thrown. In particular, if Jenkins interrupts fastlane due to a timeout, I don't want booted simulators left lying around.

Reply:
I can do that. One concern I have about that is that people actually rely on the current behavior of leaving the booted simulators lying around. I'll set that up and see what the people in charge of approving/merging say.